### PR TITLE
Add test for jsonld signatures

### LINKED ISSUE #115: Add JSON-LD Context for Element
https://github.com/decentralized-identity/element/pull/114/files#r355497681

Similar to what we have for: https://context.transmute.org/

and 

https://github.com/transmute-industries/vc-greeting-card

### PR DESCRIPTION
**Pull request by gjgd**
closes #115

### CODE REVIEW

File: packages/element-lib/src/__tests__/__fixtures__/didDocumentWithCustomContext.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "@context": "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld",
```
Comment (OR13) [inline packages/element-lib/src/__tests__/__fixtures__/didDocumentWithCustomContext.json]: This needs to be an array, with the method specific context in the second position for interop.

Comment (gjgd) [reply packages/element-lib/src/__tests__/__fixtures__/didDocumentWithCustomContext.json]: It is an array with the method specific context when you resolve it: https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld

Comment (OR13) [reply packages/element-lib/src/__tests__/__fixtures__/didDocumentWithCustomContext.json]: See the did core spec on extending context with method specific context

File: packages/element-lib/package.json [hunk 2]
```diff
@@ -72,6 +72,7 @@
   },
   "devDependencies": {
     "@transmute/eslint-config": "0.1.0",
+    "did-method-key": "^0.3.0",
```
Comment (OR13) [inline packages/element-lib/package.json]: beware that this upgrade changed the way that publicKeyIds are represented to `did:example:123#kid` from just reusing the did as the publicKeyId.

Comment (gjgd) [reply packages/element-lib/package.json]: Looks like this isn't used, forgot to remove it

**Comment by OR13** [comment]
Hmm, this is still not what I am expecting to see... I would expect the unsigned payload to look like this:

```
{
  "@context": [
    "https://w3id.org/did/v1",
    "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld"
  ],
  "id": "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q",
  "publicKey": [
    {
      "id": "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q#z6MkwPVuumvDUa5vqQsVzcubfd4YgFDenk4CdPgkwrYSeCgU",
      "type": "Ed25519VerificationKey2018",
      "controller": "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q",
      "usage": "signing",
      "publicKeyBase58": "HwEsKXfn92bTiv2oK3wkpXWYrfwoNroqwNmq7aaRiyu6"
    }
  ],
  "authentication": [
    "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q#z6MkwPVuumvDUa5vqQsVzcubfd4YgFDenk4CdPgkwrYSeCgU"
  ],
  "assertionMethod": [
    "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q#z6MkwPVuumvDUa5vqQsVzcubfd4YgFDenk4CdPgkwrYSeCgU"
  ],
  "capabilityDelegation": [
    "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q#z6MkwPVuumvDUa5vqQsVzcubfd4YgFDenk4CdPgkwrYSeCgU"
  ],
  "capabilityInvocation": [
    "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q#z6MkwPVuumvDUa5vqQsVzcubfd4YgFDenk4CdPgkwrYSeCgU"
  ],
  "keyAgreement": [
    {
      "id": "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q#zC7ELhq92KjAx6dJ8jwAiyeHKzRBbDcj7TRvpPk1cg7NGm",
      "type": "X25519KeyAgreementKey2019",
      "usage": "keyAgreement",
      "controller": "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q",
      "publicKeyBase58": "7jczTxD8MRffXFAHXgA1V7ehAunG42VxY3XMtvpH18AZ"
    }
  ]
}

```

However, I can see from the did core spec, that we have confusion over how to actually do this correctly (The spec is IMO, wrong in its current form...).... 

I believe this comment represents the correct solution (the one I would like to see):

https://github.com/w3c/did-core/pull/142#issuecomment-565681742

**Comment by OR13** [comment]
We may want to discuss this in higher bandwidth.

**Comment by gjgd** [comment]
There are a lot of ways to extend the DID method specifications with a custom JSON-LD context

1) Using an array where the custom context is referenced
```json
{
  "@context": [
    "https://w3id.org/did/v1",
    "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld"
  ],
  "id": "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q",
  ...
}
```

where "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld" resolves to 
```json
{
  "@context": {
    "@version": 1.1,

    "service": "sec:service",
    "sidetree": "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1#",

    "type": "sec:type",
    "serviceEndpoint": "sec:serviceEndpoint",
    "publicKeyHex": "sec:publicKeyHex",
    "publicKeyJwk": "sec:publicKeyJwk",
    "crv": "sec:crv",
    "kid": "sec:kid",
    "kty": "sec:kty",
    "x": "sec:x",
    "y": "sec:y",
    "usage": "sidetree:usage"
}
```

2) Using an array where the custom context is embedded
```json
{
  "@context": [
    "https://w3id.org/did/v1",
    {
      "@version": 1.1,

      "service": "sec:service",
      "sidetree": "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1#",

      "type": "sec:type",
      "serviceEndpoint": "sec:serviceEndpoint",
      "publicKeyHex": "sec:publicKeyHex",
      "publicKeyJwk": "sec:publicKeyJwk",
      "crv": "sec:crv",
      "kid": "sec:kid",
      "kty": "sec:kty",
      "x": "sec:x",
      "y": "sec:y",
      "usage": "sidetree:usage"
    }
  ],
  "id": "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q",
  ...
}
```

3) By referencing the array of context itself
```json
{
  "@context": "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld",
  "id": "did:elem:EiC_H1NwXPlS5C-JSPJJtdZpHUzbIcOW_w0dUc5jv5ZO7Q",
  ...
}
```
where "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1.jsonld" resolves to 
```json
{
  "@context": [
    "https://w3id.org/did/v1",
    {
      "@version": 1.1,

      "service": "sec:service",
      "sidetree": "https://docs.element-did.com/contexts/sidetree/sidetree-v0.1#",

      "type": "sec:type",
      "serviceEndpoint": "sec:serviceEndpoint",
      "publicKeyHex": "sec:publicKeyHex",
      "publicKeyJwk": "sec:publicKeyJwk",
      "crv": "sec:crv",
      "kid": "sec:kid",
      "kty": "sec:kty",
      "x": "sec:x",
      "y": "sec:y",
      "usage": "sidetree:usage"
    }
}
```

1) 2) 3) are equivalent from a JSON-LD processing perspective. They all achieve the same purpose of extending the did-core context with method specific vocabulary.
I think it is a bad idea to require one syntax over the others. We should be leveraging the extensive JSON-LD extensibility features instead..

**Comment by gjgd** [comment]
I recognize that [the spec](https://w3c.github.io/did-core/#contexts) says it should be an array of URIs. But I don't understand why we would deprive ourselves from using embedded contexts. I'm raising an issue about this

**Comment by OR13** [comment]
From the did core spec perspective, I think its desirable to not alter their core context, and minimize the amount of information we ask IPFS to store for sidetree (so don't embed a context object)... There are a lot of open issues about context already, you may want to just comment on https://github.com/w3c/did-core/issues/148

**Comment by OR13** [comment]
If solution 1 works, I think we should use that and move towards setting a good example for the rest of the community. They other solutions, while valid JSON-LD feel like they will be harder to get adopted.

**Comment by gjgd** [comment]
I've update the context and the example. I agree that the array of two URIs is nice, but this should not be a requirement.

More on this: https://github.com/w3c/did-core/issues/153 https://github.com/w3c/did-core/issues/148

**Pull request merged**